src/Page/Transactions/Transactions.tsx

```diff
@@ -30,23 +34,102 @@ const AssetsContainer = styled.div`
+        getAddressTxAll(address);
+      };
+    };
+  });
+
+  // Set asset options
+  const assetOptions = ['All Assets'].concat(assets.map(item => item.denom));
+
+  const getImg = (img: string | undefined) => {
+    // Convert to hash if needed
+    if (img === 'nhash') {
+      return 'hash';
+    };
+    // Check if svg exists
+    if (img && imgOptions.includes(img)) {
+      return img;
+    }
+    // Return Provenance Logo otherwise
+    return 'provenance';
+  };
+
+  const getTransactionData = () => {
+    const transactionData: typeof allTransactions = [];
+    // All Assets and All Transactions
+    if (selectedAsset === 'All Assets' && selectedTxType === 'All Transactions') {
+      allTransactions.map(t => transactionData.push(t));
+    };
+    // All Assets and Specific Transactions
+    if (selectedAsset === 'All Assets' && selectedTxType !== 'AllTransactions') {
+      allTransactions.forEach(transaction => {
+        if (transaction.status === selectedTxType.toLowerCase()) {
+          transactionData.push(transaction);
+        };
+      })
+    };
+    // Specific Assets and All Transactions
+    if (selectedAsset !== 'All Assets' && selectedTxType === 'All Transactions') {
+      allTransactions.forEach(transaction => {
+        if (transaction.denom === selectedAsset.toLowerCase()) {
+          transactionData.push(transaction);
+        };
+      })
+    };
+    // Specific Assets and Specific Transactions
+    if (selectedAsset !== 'All Assets' && selectedTxType !== 'All Transactions') {
+      allTransactions.forEach(transaction => {
+        if (transaction.denom === selectedAsset.toLowerCase() && transaction.status === selectedTxType.toLowerCase()) {
+          transactionData.push(transaction);
+        };
+      })
+    }
+    return transactionData;
+  };
+
+  // TODO: Add a message if transactionData is empty. Will add once filters are provided
   return (
     <Container>
-      <Title>Transaction Details</Title>
+      <Title>Transactions</Title>
       <Select onChange={setSelectedAsset} options={assetOptions} value={selectedAsset} />
       <Select onChange={setSelectedTxType} options={transactionOptions} value={selectedTxType} />
-      <AssetsContainer>
-        <AssetRow img="hash" name="hash" amount={{ count: '500', change: 13.63 }} />
-        <AssetRow img="usdf" name="usdf" />
-        <AssetRow img="etf" name="etf" />
-        <AssetRow img="inu" name="inu" />
-      </AssetsContainer>
+      {assetsLoading || allTransactionsLoading ? <Loading /> :
+        <AssetsContainer>
+          {getTransactionData().map(transaction => (
+            <AssetRow 
+              img={getImg(transaction.denom)}
+              name={transaction.hash} 
+              displayName={trimString(transaction?.denom === 'nhash' ? 'hash' : transaction.denom || capitalize(transaction.type), 21, 10)} 
+              amount={{ count: `fee: ${(hashFormat('hash', Number(transaction.feeAmount))).toFixed(2)} hash` }}
```

Comment:
Let's allow hashFormat to take in a string:
```export const hashFormat = (type = 'nhash', value: number | string) => {```
That should prevent needing to do any Number() functions here (gets converted into a string within hashFormat anyway)